wordpress-mobile/WordPress-iOS
# Better offline messages: full implementation

Fixes #5759 

This has been reviewed in pieces already, during hack week, but here is the full feature in one single PR. 

Expected behaviour in Comments, Notifications, Posts List, Pages List, Reader and People:
- If there is no cached data, we use NoResultsView to present a user-facing message informing that the network is down
- If there is cached data, we present an alert
- Content should reload when the network connection is back up

To test:
1. Put the device in flight mode
2. Navigate to the views mentioned
3. Exit flight mode and check that content reloads
